### PR TITLE
add pearsonr as metric

### PR DESCRIPTION
**Pull request by szha**

### CODE REVIEW

**Comment by szha** [comment]
@piiswrong

File: python/mxnet/metric.py [hunk 1]
```diff
@@ -81,12 +85,12 @@ def update_dict(self, label, pred):
         if self.output_names is not None:
             pred = [pred[name] for name in self.output_names]
         else:
-            pred = list(pred.values())
```
Comment (piiswrong) [inline python/mxnet/metric.py]: revert

File: python/mxnet/metric.py [hunk 2]
```diff
@@ -896,6 +904,60 @@ def update(self, labels, preds):
...
+
+        preds : list of `NDArray`
+            Predicted values.
+        """
+        _check_lengths_equal(labels, preds)
+        for label, pred in zip(labels, preds):
+            _check_shapes_equal(label, pred)
+            label = label.asnumpy()
+            pred = pred.asnumpy()
+            self.sum_metric += numpy.corrcoef(pred.ravel(), label.ravel())[0, 1]
```
Comment (piiswrong) [inline python/mxnet/metric.py]: what's [0, 1]

Comment (szha) [inline python/mxnet/metric.py]: corrcoef returns a pearson correlation matrix. [0][1] has the coefficient.

**Pull request merged**